File: 백준/Bronze/2884. 알람 시계/알람 시계.py
H, M = map(int, input().split())
if 0 <= H <= 23 and 0 <= M <= 59:
    if M <= 44:
        H = H - 1
        M = M + 15
        if H < 0:
            H = H + 24
        else:
            pass
    else:
        M = M - 45
    print(f"{H} {M}")
else:
    pass

File: 백준/Bronze/2884. 알람 시계/알람 시계.js
const fs = require("fs");
let [hour, minute] = fs
  .readFileSync("/dev/stdin")
  .toString()
  .split(" ")
  .map(Number);

if (0 <= hour <= 23 && 0 <= minute <= 59) {
  if (minute < 45) {
    hour -= 1;
    minute += 15;
    if (hour < 0) {
      hour += 24;
    }
  } else {
    minute -= 45;
  }
}
{
  console.log(hour, minute);
}
